File: TexasProgram/DesignSystem/Components.swift
import SwiftUI

// MARK: - Карточка

struct CardView<Content: View>: View {
    @Environment(\.colorScheme) private var scheme
    private let padding: CGFloat
    private let spacing: CGFloat
    private let cornerRadius: CGFloat
    private let content: Content

    init(padding: CGFloat = 18, spacing: CGFloat = 14, cornerRadius: CGFloat = 24, @ViewBuilder content: () -> Content) {
        self.padding = padding
        self.spacing = spacing
        self.cornerRadius = cornerRadius
        self.content = content()
    }

    private var shape: RoundedRectangle { RoundedRectangle(cornerRadius: cornerRadius, style: .continuous) }

    var body: some View {
        VStack(alignment: .leading, spacing: spacing) { content }
            .padding(padding)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(Theme.surface(scheme), in: shape)
            .overlay(shape.strokeBorder(Theme.hairline(scheme), lineWidth: 1))
    }
}

/// Карточка с подсвеченной градиентной рамкой — для «следующего» элемента.
/// Свечение статичное: анимация радиуса тени пересобирала слой на каждом кадре.
struct HighlightCard<Content: View>: View {
    @Environment(\.colorScheme) private var scheme
    private let gradient: LinearGradient
    private let glow: Color
    private let content: Content

    init(gradient: LinearGradient = Theme.accentGradient, glow: Color = Theme.accent, @ViewBuilder content: () -> Content) {
        self.gradient = gradient
        self.glow = glow
        self.content = content()
    }

    private var shape: RoundedRectangle { RoundedRectangle(cornerRadius: 26, style: .continuous) }

    var body: some View {
        VStack(alignment: .leading, spacing: 14) { content }
            .padding(20)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(Theme.surface(scheme), in: shape)
            .overlay(shape.strokeBorder(gradient, lineWidth: 1.6))
            .shadow(color: glow.opacity(scheme == .dark ? 0.28 : 0.16), radius: 14, y: 6)
    }
}

// MARK: - Кнопки

struct PressableStyle: ButtonStyle {
    var scale: CGFloat = 0.97

    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .scaleEffect(configuration.isPressed ? scale : 1)
            .opacity(configuration.isPressed ? 0.9 : 1)
            .animation(.spring(response: 0.3, dampingFraction: 0.7), value: configuration.isPressed)
    }
}

extension ButtonStyle where Self == PressableStyle {
    static var pressable: PressableStyle { PressableStyle() }
}

struct GradientButtonStyle: ButtonStyle {
    var gradient: LinearGradient = Theme.accentGradient
    var glow: Color = Theme.accent

    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .font(.headline)
            .foregroundStyle(.white)
            .padding(.vertical, 16)
            .frame(maxWidth: .infinity)
            .background(gradient, in: RoundedRectangle(cornerRadius: 18, style: .continuous))
            .shadow(color: glow.opacity(configuration.isPressed ? 0.15 : 0.35), radius: configuration.isPressed ? 6 : 16, y: 8)
            .scaleEffect(configuration.isPressed ? 0.975 : 1)
            .animation(.spring(response: 0.3, dampingFraction: 0.7), value: configuration.isPressed)
    }
}

extension ButtonStyle where Self == GradientButtonStyle {
    static var gradientProminent: GradientButtonStyle { GradientButtonStyle() }
}

// MARK: - Кольцо прогресса

struct RingProgress<Label: View>: View {
    private let value: Double
    private let lineWidth: CGFloat
    private let gradient: LinearGradient
    private let label: Label
    @Environment(\.accessibilityReduceMotion) private var reduceMotion
    @State private var animated: Double = 0

    init(value: Double, lineWidth: CGFloat = 14, gradient: LinearGradient = Theme.accentGradient, @ViewBuilder label: () -> Label) {
        self.value = value.isFinite ? min(max(value, 0), 1) : 0
        self.lineWidth = lineWidth
        self.gradient = gradient
        self.label = label()
    }

    var body: some View {
        ZStack {
            Circle().stroke(Color.primary.opacity(0.09), style: StrokeStyle(lineWidth: lineWidth, lineCap: .round))
            Circle()
                .trim(from: 0, to: animated)
                .stroke(gradient, style: StrokeStyle(lineWidth: lineWidth, lineCap: .round))
                .rotationEffect(.degrees(-90))
            label
        }
        .onAppear {
            if reduceMotion { animated = value } else {
                withAnimation(.easeOut(duration: 1.0).delay(0.15)) { animated = value }
            }
        }
        .onChange(of: value) { _, newValue in
            withAnimation(Motion.maybe(Motion.card, reduce: reduceMotion)) { animated = newValue }
        }
    }
}

// MARK: - Линейный индикатор

struct LineMeter: View {
    let value: Double
    var gradient: LinearGradient = Theme.accentGradient
    var height: CGFloat = 8
    @Environment(\.accessibilityReduceMotion) private var reduceMotion
    @State private var animated: Double = 0

    var body: some View {
        GeometryReader { proxy in
            ZStack(alignment: .leading) {
                Capsule().fill(Color.primary.opacity(0.09))
                Capsule()
                    .fill(gradient)
                    .frame(width: max(0, proxy.size.width * min(max(animated, 0), 1)))
            }
        }
        .frame(height: height)
        .onAppear {
            if reduceMotion { animated = value } else {
                withAnimation(.easeOut(duration: 0.8).delay(0.1)) { animated = value }
            }
        }
        .onChange(of: value) { _, newValue in
            withAnimation(Motion.maybe(Motion.card, reduce: reduceMotion)) { animated = newValue }
        }
    }
}

// MARK: - Бейдж

struct TagBadge: View {
    let text: String
    var systemImage: String?
    var gradient: LinearGradient = Theme.accentGradient

    var body: some View {
        HStack(spacing: 5) {
            if let systemImage { Image(systemName: systemImage).font(.caption2.weight(.bold)) }
            Text(text).font(.caption.weight(.semibold))
        }
        .padding(.horizontal, 10)
        .padding(.vertical, 5)
        .foregroundStyle(.white)
        .background(gradient, in: Capsule())
    }
}

struct OutlineBadge: View {
    let text: String
    var tint: Color = Theme.accent

    var body: some View {
        Text(text)
            .font(.caption.weight(.semibold))
            .padding(.horizontal, 10)
            .padding(.vertical, 5)
            .foregroundStyle(tint)
            .background(tint.opacity(0.14), in: Capsule())
            .overlay(Capsule().strokeBorder(tint.opacity(0.35), lineWidth: 1))
    }
}

// MARK: - Заголовок экрана

struct ScreenHeader: View {
    let eyebrow: String
    let title: String
    var subtitle: String?

    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text(eyebrow.uppercased())
                .font(.caption.weight(.bold))
                .tracking(1.4)
                .foregroundStyle(Theme.accentGradient)
            Text(title)
                .font(.system(size: 32, weight: .bold, design: .rounded))
            if let subtitle {
                Text(subtitle).font(.subheadline).foregroundStyle(.secondary)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}

// MARK: - Появление и прокрутка

/// Каскадное появление. Без `blur`: размытие рендерится вне экрана и на списке
/// из десятка карточек съедает кадры ровно в момент открытия экрана.
private struct AppearIn: ViewModifier {
    let index: Int
    @Environment(\.accessibilityReduceMotion) private var reduceMotion
    @State private var shown = false

    func body(content: Content) -> some View {
        content
            .opacity(shown ? 1 : 0)
            .offset(y: shown ? 0 : 14)
            .onAppear {
                guard !shown else { return }
                if reduceMotion {
                    shown = true
                } else {
                    let delay = Double(min(index, 6)) * 0.045
                    withAnimation(.spring(response: 0.5, dampingFraction: 0.86).delay(delay)) { shown = true }
                }
            }
    }
}

/// Затухание карточек у края экрана. Только opacity и scale — обе операции
/// выполняет композитор, в отличие от размытия.
private struct SoftScroll: ViewModifier {
    @Environment(\.accessibilityReduceMotion) private var reduceMotion

    @ViewBuilder
    func body(content: Content) -> some View {
        if reduceMotion {
            content
        } else {
            content.scrollTransition(.interactive, axis: .vertical) { view, phase in
                view
                    .opacity(phase.isIdentity ? 1 : 0.35)
                    .scaleEffect(phase.isIdentity ? 1 : 0.95)
            }
        }
    }
}

extension View {
    /// Каскадное появление элемента с задержкой по индексу.
    func appearIn(_ index: Int) -> some View { modifier(AppearIn(index: index)) }

    /// Мягкое затухание и уменьшение при уходе за край экрана.
    func softScroll() -> some View { modifier(SoftScroll()) }
}

File: android/app/src/main/java/com/texasprogram/app/ui/Components.kt
package com.texasprogram.app.ui

import android.view.HapticFeedbackConstants
import androidx.compose.animation.core.Animatable
import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.combinedClickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.interaction.collectIsPressedAsState
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyListState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.BasicTextField
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.KeyboardArrowLeft
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.composed
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.SolidColor
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.graphics.graphicsLayer
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalView
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.foundation.Canvas
import kotlinx.coroutines.delay
import kotlin.math.abs

// MARK: - Модификаторы

/// Каскадное появление: прозрачность и сдвиг вверх с задержкой по индексу.
fun Modifier.appearIn(index: Int): Modifier = composed {
    val progress = remember { Animatable(0f) }
    LaunchedEffect(Unit) {
        delay(index.coerceAtMost(12) * Motion.STAGGER_MS.toLong())
        progress.animateTo(1f, Motion.appear())
    }
    graphicsLayer {
        alpha = progress.value
        translationY = (1f - progress.value) * 18.dp.toPx()
    }
}

/// Нажатие с пружинным сжатием — аналог PressableStyle из SwiftUI.
fun Modifier.pressable(enabled: Boolean = true, onClick: () -> Unit): Modifier = composed {
    val interaction = remember { MutableInteractionSource() }
    val pressed by interaction.collectIsPressedAsState()
    val scale by animateFloatAsState(if (pressed) 0.97f else 1f, Motion.snappy(), label = "press")
    graphicsLayer { scaleX = scale; scaleY = scale }
        .clickable(
            interactionSource = interaction,
            indication = null,
            enabled = enabled,
            onClick = onClick
        )
}

/// Тап и долгое нажатие с той же анимацией. Отдельная функция нужна потому,
/// что `clickable` и `combinedClickable` — разные модификаторы.
fun Modifier.combinedPressable(onClick: () -> Unit, onLongClick: (() -> Unit)? = null): Modifier = composed {
    val interaction = remember { MutableInteractionSource() }
    val pressed by interaction.collectIsPressedAsState()
    val scale by animateFloatAsState(if (pressed) 0.97f else 1f, Motion.snappy(), label = "press")
    graphicsLayer { scaleX = scale; scaleY = scale }
        .combinedClickable(
            interactionSource = interaction,
            indication = null,
            onLongClick = onLongClick,
            onClick = onClick
        )
}

/// Мягкое затухание у краёв экрана. Состояние читается в фазе отрисовки,
/// поэтому прокрутка не вызывает рекомпозицию.
fun Modifier.softScroll(state: LazyListState, index: Int): Modifier = graphicsLayer {
    val info = state.layoutInfo
    val item = info.visibleItemsInfo.firstOrNull { it.index == index } ?: return@graphicsLayer
    val viewportStart = info.viewportStartOffset.toFloat()
    val viewportEnd = info.viewportEndOffset.toFloat()
    val itemCenter = item.offset + item.size / 2f
    val viewportCenter = (viewportStart + viewportEnd) / 2f
    val half = (viewportEnd - viewportStart) / 2f
    if (half <= 0f) return@graphicsLayer
    val distance = (abs(itemCenter - viewportCenter) / half).coerceIn(0f, 1f)
    val fade = ((distance - 0.72f) / 0.28f).coerceIn(0f, 1f)
    alpha = 1f - fade * 0.7f
    val shrink = 1f - fade * 0.06f
    scaleX = shrink
    scaleY = shrink
}

@Composable
fun rememberHaptics(): () -> Unit {
    val view = LocalView.current
    return remember(view) {
        { view.performHapticFeedback(HapticFeedbackConstants.CONTEXT_CLICK) }
    }
}

// MARK: - Карточки

@Composable
fun CardView(
    modifier: Modifier = Modifier,
    padding: Dp = 18.dp,
    spacing: Dp = 14.dp,
    corner: Dp = 24.dp,
    content: @Composable ColumnScope.() -> Unit
) {
    Column(
        modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(corner))
            .background(Theme.surface)
            .border(1.dp, Theme.hairlineGradient, RoundedCornerShape(corner))
            .padding(padding),
        verticalArrangement = Arrangement.spacedBy(spacing),
        content = content
    )
}

/// Карточка с градиентной рамкой — для «следующего» элемента.
@Composable
fun HighlightCard(
    modifier: Modifier = Modifier,
    gradient: Brush = Theme.accentGradient,
    content: @Composable ColumnScope.() -> Unit
) {
    Column(
        modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(26.dp))
            .background(Theme.surface)
            .border(1.6.dp, gradient, RoundedCornerShape(26.dp))
            .padding(20.dp),
        verticalArrangement = Arrangement.spacedBy(14.dp),
        content = content
    )
}

// MARK: - Индикаторы

@Composable
fun RingProgress(
    value: Float,
    modifier: Modifier = Modifier,
    lineWidth: Dp = 12.dp,
    gradient: Brush = Theme.accentGradient,
    content: @Composable () -> Unit
) {
    val animated = remember { Animatable(0f) }
    LaunchedEffect(value) {
        animated.animateTo(value.coerceIn(0f, 1f), Motion.card())
    }
    Box(modifier, contentAlignment = Alignment.Center) {
        Canvas(Modifier.matchParentSize()) {
            val stroke = lineWidth.toPx()
            val inset = stroke / 2f
            val arcSize = Size(size.width - stroke, size.height - stroke)
            drawArc(
                color = Color.White.copy(alpha = 0.09f),
                startAngle = -90f,
                sweepAngle = 360f,
                useCenter = false,
                topLeft = androidx.compose.ui.geometry.Offset(inset, inset),
                size = arcSize,
                style = Stroke(width = stroke, cap = androidx.compose.ui.graphics.StrokeCap.Round)
            )
            if (animated.value > 0.001f) {
                drawArc(
                    brush = gradient,
                    startAngle = -90f,
                    sweepAngle = 360f * animated.value,
                    useCenter = false,
                    topLeft = androidx.compose.ui.geometry.Offset(inset, inset),
                    size = arcSize,
                    style = Stroke(width = stroke, cap = androidx.compose.ui.graphics.StrokeCap.Round)
                )
            }
        }
        content()
    }
}

@Composable
fun LineMeter(
    value: Float,
    modifier: Modifier = Modifier,
    gradient: Brush = Theme.accentGradient,
    height: Dp = 8.dp
) {
    val animated = remember { Animatable(0f) }
    LaunchedEffect(value) { animated.animateTo(value.coerceIn(0f, 1f), Motion.card()) }
    Box(
        modifier
            .fillMaxWidth()
            .height(height)
            .clip(CircleShape)
            .background(Color.White.copy(alpha = 0.09f))
    ) {
        Box(
            Modifier
                .fillMaxWidth(animated.value)
                .height(height)
                .clip(CircleShape)
                .background(gradient)
        )
    }
}

// MARK: - Бейджи

@Composable
fun TagBadge(
    text: String,
    modifier: Modifier = Modifier,
    icon: ImageVector? = null,
    gradient: Brush = Theme.accentGradient
) {
    Row(
        modifier
            .clip(CircleShape)
            .background(gradient)
            .padding(horizontal = 10.dp, vertical = 5.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(5.dp)
    ) {
        if (icon != null) {
            Icon(icon, contentDescription = null, tint = Color.White, modifier = Modifier.size(12.dp))
        }
        Text(text, color = Color.White, fontSize = 12.sp, fontWeight = FontWeight.SemiBold)
    }
}

@Composable
fun OutlineBadge(text: String, modifier: Modifier = Modifier, tint: Color = Theme.accent) {
    Text(
        text,
        color = tint,
        fontSize = 12.sp,
        fontWeight = FontWeight.SemiBold,
        modifier = modifier
            .clip(CircleShape)
            .background(tint.copy(alpha = 0.14f))
            .border(1.dp, tint.copy(alpha = 0.35f), CircleShape)
            .padding(horizontal = 10.dp, vertical = 5.dp)
    )
}

// MARK: - Текст

@Composable
fun ScreenHeader(eyebrow: String, title: String, subtitle: String? = null, modifier: Modifier = Modifier) {
    Column(modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(6.dp)) {
        GradientText(eyebrow.uppercase(), fontSize = 12.sp, weight = FontWeight.Bold, letterSpacing = 1.4f)
        Text(title, color = Theme.textPrimary, fontSize = 32.sp, fontWeight = FontWeight.Bold)
        if (subtitle != null) {
            Text(subtitle, color = Theme.textSecondary, fontSize = 14.sp)
        }
    }
}

@Composable
fun GradientText(
    text: String,
    modifier: Modifier = Modifier,
    fontSize: androidx.compose.ui.unit.TextUnit = 16.sp,
    weight: FontWeight = FontWeight.Bold,
    gradient: Brush = Theme.accentGradient,
    letterSpacing: Float = 0f,
    textAlign: TextAlign? = null
) {
    Text(
        text = text,
        modifier = modifier,
        fontSize = fontSize,
        fontWeight = weight,
        letterSpacing = letterSpacing.sp,
        textAlign = textAlign,
        style = androidx.compose.ui.text.TextStyle(brush = gradient)
    )
}

// MARK: - Навигация

/// Шапка как в iOS: шеврон слева, заголовок по центру.
@Composable
fun TopBar(
    title: String,
    modifier: Modifier = Modifier,
    onBack: (() -> Unit)? = null,
    trailing: @Composable (() -> Unit)? = null
) {
    Box(
        modifier
            .fillMaxWidth()
            .height(44.dp)
    ) {
        if (onBack != null) {
            Box(
                Modifier
                    .align(Alignment.CenterStart)
                    .size(40.dp)
                    .clip(CircleShape)
                    .pressable(onClick = onBack),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    Icons.Filled.KeyboardArrowLeft,
                    contentDescription = "Назад",
                    tint = Theme.textPrimary,
                    modifier = Modifier.size(28.dp)
                )
            }
        }
        Text(
            title,
            color = Theme.textPrimary,
            fontSize = 17.sp,
            fontWeight = FontWeight.SemiBold,
            maxLines = 1,
            modifier = Modifier.align(Alignment.Center)
        )
        if (trailing != null) {
            Box(Modifier.align(Alignment.CenterEnd)) { trailing() }
        }
    }
}

// MARK: - Сегментный переключатель

/// Нейтральный серый, как системный segmented control в iOS,
/// а не акцентный градиент.
@Composable
fun SegmentedControl(
    options: List<String>,
    selectedIndex: Int,
    modifier: Modifier = Modifier,
    onSelect: (Int) -> Unit
) {
    Row(
        modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(10.dp))
            .background(Color.White.copy(alpha = 0.07f))
            .padding(3.dp),
        horizontalArrangement = Arrangement.spacedBy(3.dp)
    ) {
        options.forEachIndexed { index, title ->
            val selected = index == selectedIndex
            Box(
                Modifier
                    .weight(1f)
                    .clip(RoundedCornerShape(8.dp))
                    .background(if (selected) Color.White.copy(alpha = 0.18f) else Color.Transparent)
                    .pressable { onSelect(index) }
                    .padding(vertical = 9.dp),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    title,
                    color = Theme.textPrimary,
                    fontSize = 14.sp,
                    fontWeight = if (selected) FontWeight.SemiBold else FontWeight.Normal,
                    maxLines = 1
                )
            }
        }
    }
}

// MARK: - Поле веса

/// Строка веса как в iOS: подпись, минус, редактируемое значение, плюс.
@Composable
fun WeightStepper(
    label: String,
    text: String,
    modifier: Modifier = Modifier,
    step: Double = 2.5,
    onTextChange: (String) -> Unit
) {
    val current = text.replace(",", ".").toDoubleOrNull() ?: 0.0

    fun apply(delta: Double) {
        val next = (current + delta).coerceAtLeast(0.0)
        onTextChange(if (next == kotlin.math.floor(next)) next.toInt().toString() else next.toString())
    }

    Row(modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
        Text(
            label,
            color = Theme.textPrimary,
            fontSize = 15.sp,
            maxLines = 1,
            modifier = Modifier.weight(1f, fill = false)
        )
        Spacer(Modifier.weight(1f))
        StepCircle("−") { apply(-step) }
        Spacer(Modifier.width(8.dp))
        Row(
            Modifier
                .clip(RoundedCornerShape(12.dp))
                .background(Color.White.copy(alpha = 0.06f))
                .padding(horizontal = 12.dp, vertical = 8.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            BasicTextField(
                value = text,
                onValueChange = { new -> onTextChange(new.filter { it.isDigit() || it == '.' || it == ',' }) },
                singleLine = true,
                textStyle = androidx.compose.ui.text.TextStyle(
                    color = Theme.textPrimary,
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    textAlign = TextAlign.End
                ),
                cursorBrush = SolidColor(Theme.accent),
                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Decimal),
                modifier = Modifier.width(52.dp)
            )
            Spacer(Modifier.width(4.dp))
            Text("кг", color = Theme.textSecondary, fontSize = 12.sp, fontWeight = FontWeight.SemiBold)
        }
        Spacer(Modifier.width(8.dp))
        StepCircle("+") { apply(step) }
    }
}

@Composable
private fun StepCircle(symbol: String, onClick: () -> Unit) {
    Box(
        Modifier
            .size(34.dp)
            .clip(CircleShape)
            .background(Theme.accent.copy(alpha = 0.16f))
            .pressable(onClick = onClick),
        contentAlignment = Alignment.Center
    ) {
        Text(symbol, color = Theme.accent, fontSize = 17.sp, fontWeight = FontWeight.Black)
    }
}

// MARK: - Кнопки

@Composable
fun GradientButton(
    text: String,
    modifier: Modifier = Modifier,
    icon: ImageVector? = null,
    gradient: Brush = Theme.accentGradient,
    enabled: Boolean = true,
    onClick: () -> Unit
) {
    Row(
        modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(18.dp))
            .background(gradient)
            .alphaIf(!enabled)
            .pressable(enabled = enabled, onClick = onClick)
            .padding(vertical = 16.dp),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically
    ) {
        if (icon != null) {
            Icon(icon, contentDescription = null, tint = Color.White, modifier = Modifier.size(20.dp))
            Text("  ", fontSize = 16.sp)
        }
        Text(text, color = Color.White, fontSize = 16.sp, fontWeight = FontWeight.SemiBold)
    }
}

@Composable
fun SecondaryButton(
    text: String,
    modifier: Modifier = Modifier,
    icon: ImageVector? = null,
    tint: Color = Theme.accent,
    onClick: () -> Unit
) {
    Row(
        modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(tint.copy(alpha = 0.12f))
            .border(1.dp, tint.copy(alpha = 0.35f), RoundedCornerShape(12.dp))
            .pressable(onClick = onClick)
            .padding(vertical = 11.dp),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically
    ) {
        if (icon != null) {
            Icon(icon, contentDescription = null, tint = tint, modifier = Modifier.size(16.dp))
            Text("  ", fontSize = 13.sp)
        }
        Text(text, color = tint, fontSize = 13.sp, fontWeight = FontWeight.SemiBold)
    }
}

private fun Modifier.alphaIf(dim: Boolean): Modifier = if (dim) this.graphicsLayer { alpha = 0.5f } else this
